aspnetcore/fundamentals/servers/aspnet-core-module.md

```diff
@@ -4,38 +4,107 @@ author: rick-anderson
+
+The ASP.NET Core Module:
+
+* Performs app initialization.
+  * Loads the [CoreCLR](/dotnet/standard/glossary#coreclr).
+  * Calls `Program.Main`.
+* Handles the lifetime of the IIS native request.
+
+The following diagram illustrates the relationship between IIS, the ASP.NET Core Module, and an app hosted in-process:
+
+![ASP.NET Core Module](aspnet-core-module/_static/ancm-inprocess.png)
+
+A request arrives from the web to the kernel-mode HTTP.sys driver. The driver routes the request to IIS on the website's configured port, usually 80 (HTTP) or 443 (HTTPS). The module manages the native request context and converts it into a managed request <xref:Microsoft.AspNetCore.Http.HttpContext> for a custom <xref:Microsoft.AspNetCore.Hosting.Server.IServer> implementation, `IISHttpServer`.
+
+After `IISHttpServer` picks up the request, the request is pushed into the ASP.NET Core middleware pipeline. The middleware pipeline handles the request and passes it on as an `HttpContext` instance to the app's logic. The app's response is passed back to IIS, which pushes it back out to the HTTP client that initiated the request.
+
+### Out-of-process hosting model
+
+Because ASP.NET Core apps run in a process separate from the IIS worker process, the module handles process management. The module starts the process for the ASP.NET Core app when the first request arrives and restarts the app if it crashes. This is essentially the same behavior as seen with apps that run in-process that are managed by the [Windows Process Activation Service (WAS)](/iis/manage/provisioning-and-managing-iis/features-of-the-windows-process-activation-service-was).
+
+The following diagram illustrates the relationship between IIS, the ASP.NET Core Module, and an app hosted out-of-process:
+
+![ASP.NET Core Module](aspnet-core-module/_static/ancm-outofprocess.png)
+
+Requests arrive from the web to the kernel-mode HTTP.sys driver. The driver routes the requests to IIS on the website's configured port, usually 80 (HTTP) or 443 (HTTPS). The module forwards the requests to Kestrel on a random port for the app, which isn't port 80/443.
+
+The module specifies the port via an environment variable at startup, and the IIS Integration Middleware configures the server to listen on `http://localhost:{port}`. Additional checks are performed, and requests that don't originate from the module are rejected. The module doesn't support HTTPS forwarding, so requests are forwarded over HTTP even if received by IIS over HTTPS.
+
+After Kestrel picks up the request from the module, the request is pushed into the ASP.NET Core middleware pipeline. The middleware pipeline handles the request and passes it on as an `HttpContext` instance to the app's logic. The app's response is passed back to IIS, which pushes it back out to the HTTP client that initiated the request.
```

Comment:
IIS Integration also adds middleware that update Request.Scheme, Connection.RemoteIP, and request.PathBase to account for the forwarding that has taken place.